cmd/devtools/run.go

```diff
@@ -11,49 +13,106 @@ import (
 	"github.com/spf13/cobra"
 )
 
+var IsRunLocal bool
+var IsRunRemote bool
+
+type RunConfiguration int
+
+const (
+	Local RunConfiguration = iota
+	Remote
+	Unknown
+)
+
 // runCmd represents the run command
 var runCmd = &cobra.Command{
 	Use:   "run",
 	Short: "Run all the Astria services locally.",
 	Long:  `Run all the Astria services locally. This will start the sequencer, cometbft, composer, and conductor.`,
-	Run: func(cmd *cobra.Command, args []string) {
-		runall()
-	},
+	Run:   runRun,
 }
 
 func init() {
 	devCmd.AddCommand(runCmd)
+	runCmd.Flags().StringP("instance", "i", DefaultInstanceName, "Choose where the local-dev-astria directory will be created.")
+	runCmd.Flags().BoolVarP(&IsRunLocal, "local", "l", false, "Run the Astria stack using a locally running sequencer.")
+	runCmd.Flags().BoolVarP(&IsRunRemote, "remote", "r", false, "Run the Astria stack using a remote sequencer.")
+	runCmd.MarkFlagsMutuallyExclusive("local", "remote")
 }
 
-func runall() {
+func runRun(c *cobra.Command, args []string) {
 	ctx := cmd.RootCmd.Context()
 
+	instance := c.Flag("instance").Value.String()
+	IsInstanceNameValidOrPanic(instance)
+
 	homePath, err := os.UserHomeDir()
 	if err != nil {
 		log.WithError(err).Error("Error getting home dir")
 		panic(err)
 	}
 	defaultDir := filepath.Join(homePath, ".astria")
+	instanceDir := filepath.Join(defaultDir, instance)
 
+	var runners []processrunner.ProcessRunner
+	runConfiguration, err := determineRunConfiguration()
+	if err != nil {
+		log.WithError(err).Error("Error determining run configuration")
+		panic(err)
+	}
+	switch runConfiguration {
+	case Local:
+		runners = runLocal(ctx, instanceDir)
+	case Remote:
+		runners = runRemote(ctx, instanceDir)
+	case Unknown:
+		// this should never happen
+		err := fmt.Errorf("Unknown run configuration encoutered.")
+		panic(err)
+	}
+
+	// create and start ui app
+	app := ui.NewApp(runners)
+	app.Start()
+}
+
+func determineRunConfiguration() (RunConfiguration, error) {
```

Comment:
Do we need this? What if we just put the logic below in the case statements above?

Reply:
```
    var runners []processrunner.ProcessRunner
	switch {
	case !IsRunLocal && !IsRunRemote:
		log.Info("No --local or --remote flag provided. Defaulting to --local.")
		runners = runLocal(ctx, instanceDir)
	case IsRunLocal:
		log.Info("--local flag provided. Running local sequencer.")
		runners = runLocal(ctx, instanceDir)
	case IsRunRemote:
		log.Info("--remote flag provided. Connecting to remote sequencer.")
		runners = runRemote(ctx, instanceDir)
	default:
		err := fmt.Errorf("unknown run configuration encoutered")
		panic(err)
	}
```